src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -61,7 +61,15 @@ func parseTime(r *http.Request, key string) (time.Time, error) {
 // nolint: unparam
 func parseDuration(r *http.Request, key string) (time.Duration, error) {
 	if d := r.FormValue(key); d != "" {
-		return time.ParseDuration(d)
+		value, firstErr := time.ParseDuration(d)
+		if firstErr != nil {
+			// Try parsing as an integer value specifying seconds, the Prometheus default
+			if seconds, err := strconv.ParseInt(d, 10, 64); err == nil {
+				value = time.Duration(seconds) * time.Second
```

Comment:
nit: `return time.Duration(seconds) * time.Second, nil`?

Reply:
Good call, also refactored the initial form value if/else too.